### PR TITLE
Frontend: Components: Added separate component to display Env Variables

### LINKED ISSUE #4356: Feature to show/hide value of secret
## Describe the solution you'd like
In open lens, show/hide is supported for secret value.
Image

And also, additional label for displaying name of the secret, rather than just displaying the key of that secret only.

## What users will benefit from this feature?
In-Cluster users

## Are you able to implement this feature?
Unfortunately, no.

### PR DESCRIPTION
**Pull request by mudit06mah**
## Summary:

This PR adds component to display environment variables.

## Related Issue:

Partially Fixes #4356: (PR does not add an icon to hide/show the secrets inside the pod details section itself,without going to the secrets page).

## Changes:

- Created new component file `EnvVarGrid.tsx`
- Created new storybook file: `EnvVarGrid.stories.tsx`
- Added function: `EnvVarGrid()` to display secrets like Secret Key, Config Map Key, etc.

## Screenshot:

image

### CODE REVIEW

**Comment by mudit06mah** [comment]
This PR passes all tests and is ready to be merged.

**Comment by mudit06mah** [comment]
It appears I was unaware of i18n translations, the PR should be ready to merge now:
image

I apologize for the overlook.

**Comment by mudit06mah** [comment]
@illume I have made all the changes instructed by copilot and changed the commit message as well, this PR should be ready to merge :)

File: frontend/src/components/common/Resource/EnvVarGrid.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Meta, StoryFn } from '@storybook/react';
+import { TestContext } from '../../../test';
+import { EnvVarGrid } from './EnvVarGrid';
+
+export default {
+  title: 'Resource/EnvVarDisplay',
```
Comment (illume) [inline frontend/src/components/common/Resource/EnvVarGrid.stories.tsx]: Please remove the title attribute all together, it's not needed now with storybook. Storybook can figure out the title from the component now.

(The EnvVarDisaplay name is wrong)

File: frontend/src/components/common/Resource/__snapshots__/EnvVarGrid.PlainValues.stories.storyshot [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+          class="MuiTypography-root MuiTypography-body1 css-no9i7w-MuiTypography-root"
+        >
+          DEBUG
+          : 
+          true
+        </span>
+      </div>
+    </div>
+  </div>
+</body>
```
[the remark this answers is not in the document]
Comment (illume) [reply frontend/src/components/common/Resource/__snapshots__/EnvVarGrid.PlainValues.stories.storyshot:26]: This is related to the story title, but there is another open comment for this one. So resolving this comment.

File: frontend/src/components/common/Resource/EnvVarDisplay.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        </Typography>
+      );
+    }
+
+    //FieldRef:
+    if (envVar.valueFrom?.fieldRef) {
+      const { fieldPath } = envVar.valueFrom.fieldRef;
+      return (
+        <Typography component="span" sx={envEntryStyle} key={envVar.name}>
+          {envVar.name}:FieldRef ({fieldPath})
```
[the remark this answers is not in the document]
Comment (illume) [reply frontend/src/components/common/Resource/EnvVarDisplay.tsx]: This is fixed now. Thanks

File: frontend/src/components/common/Resource/EnvVarGrid.tsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    if (envVar.valueFrom?.configMapKeyRef) {
+      const { name: cmName, key: cmKey } = envVar.valueFrom.configMapKeyRef;
+      const secretUrl = `/c/${cluster}/secrets/${namespace}/${cmName}`;
+      return (
+        <Typography>
+          {envVar.name}:{' '}
+          <Link to={secretUrl} style={{ textDecoration: 'underline', fontWeight: 'bold' }}>
+            ConfigMap: {cmName} (Key: {cmKey})
+          </Link>
+        </Typography>
```
[the remark this answers is not in the document]
Comment (illume) [reply frontend/src/components/common/Resource/EnvVarGrid.tsx]: Please consider this?

File: frontend/src/components/common/Resource/EnvVarGrid.tsx [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+import { useTranslation } from 'react-i18next';
+import Link from '../Link';
+
+interface EnvVarGridProps {
+  envVars: any[];
+  namespace: string;
+  cluster: string;
+}
+
+export function EnvVarGrid(props: EnvVarGridProps) {
```
[the remark this answers is not in the document]
Comment (illume) [reply frontend/src/components/common/Resource/EnvVarGrid.tsx:32]: There's another comment open for this issue. So resolving this one.

File: frontend/src/components/common/Resource/EnvVarGrid.tsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+  const { envVars = [], namespace, cluster } = props;
+  const { t } = useTranslation();
+  const [expanded, setExpanded] = React.useState(false);
+  const defaultNumShown = 20;
+
+  const envEntryStyle = (theme: Theme) => ({
```
Comment (sniok) [inline frontend/src/components/common/Resource/EnvVarGrid.tsx]: this should be a styled component at this point

```
const EnvEntry = styled(Typography)({
  //
})
```

docs https://v5.mui.com/system/styled/

**Comment by mudit06mah** [comment]
@illume @sniok I have made all the changes made by copilot and the maintainers, updated the commit message and PR description appropriately, Can you please review this again?

File: frontend/src/components/common/Resource/EnvVarGrid.tsx [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    borderRadius: theme.shape.borderRadius + 'px',
+    backgroundColor: theme.palette.background.muted,
+    border: '1px solid',
+    borderColor: theme.palette.divider,
+    fontSize: theme.typography.pxToRem(14),
+    padding: '4px 8px',
+    marginRight: theme.spacing(1),
+    whiteSpace: 'nowrap',
+    display: 'inline-block',
+  }));
```
[the remark this answers is not in the document]
Comment (illume) [reply frontend/src/components/common/Resource/EnvVarGrid.tsx]: Yes, please move it out.

File: frontend/src/components/common/Resource/EnvVarGrid.tsx [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+          onClick={() => setExpanded(!expanded)}
+          size="small"
+          startIcon={<Icon icon={expanded ? 'mdi:menu-up' : 'mdi:menu-down'} />}
+        >
+          {!expanded
+            ? t('translation|Show all environment variables (+{{count}} more)', {
+                count: envVars.length - defaultNumShown,
+              })
+            : t('translation|Show fewer')}
+        </Button>
```
[the remark this answers is not in the document]
Comment (illume) [reply frontend/src/components/common/Resource/EnvVarGrid.tsx:175]: Seems like a valid point.

[the remark this answers is not in the document]
Comment (illume) [reply frontend/src/components/common/Resource/EnvVarGrid.tsx:175]: Yes, this is a valid point too.

File: frontend/src/components/common/Resource/EnvVarGrid.tsx [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+      const { name: cmName, key: cmKey } = envVar.valueFrom.configMapKeyRef;
+      const configMapUrl = `/c/${cluster}/configmaps/${namespace}/${cmName}`;
+      return (
+        <EnvEntry component="span" key={envVar.name}>
+          {envVar.name}:{' '}
+          <Link to={configMapUrl} style={{ textDecoration: 'underline', fontWeight: 'bold' }}>
+            ConfigMap: {cmName} (Key: {cmKey})
+          </Link>
+        </EnvEntry>
+      );
```
[the remark this answers is not in the document]
Comment (illume) [reply frontend/src/components/common/Resource/EnvVarGrid.tsx]: You could add a storybook state to make sure this case is handled.

**Comment by mudit06mah** [comment]
@illume  I have made all the changes mentioned by you and copilot, and also added story book cases for missing name, Can you please review this again?

**Comment by mudit06mah** [comment]
@illume @sniok Can you please review this again?

**Comment by mudit06mah** [comment]
@illume @sniok Can you please review this? 👾

**Comment by illume** [comment]
@mudit06mah this would be a bit easier to review with the summary section in the PR description filled out. Also with the “steps to test “ added.

**Comment by mastermaxx03** [comment]
@illume @mudit06mah I tested this PR locally on macOS - all 4 stories look good. 
Secret and ConfigMap references show up as clickable links, 
edge cases don't crash, and the show/hide toggle works fine.
This is working as intended.

**Comment by mudit06mah** [comment]
@mastermaxx03 Thank You for reviewing!
It appears that I overlooked @illume's comment about adding steps to test, would like to apologize for that.

**Comment by illume** [comment]
Thanks @mastermaxx03 for the review
